azurerm/resource_arm_automation_schedule.go

```diff
@@ -0,0 +1,317 @@
+		firstRunDayOfMonth = closestValidStartTime.Day()
+	}
+
+	switch freq {
+	case automation.Hour:
+		validStartTime := time.Date(closestValidStartTime.Year(), closestValidStartTime.Month(), closestValidStartTime.Day(), closestValidStartTime.Hour(), firstRunMinute, firstRunSec, 0, time.UTC)
+		if firstRunMinute <= closestValidStartTime.Minute() {
+			validStartTime.Add(time.Duration(1) * time.Hour)
+		}
+
+		return validStartTime
+
+	case automation.Day:
+		validStartTime := time.Date(closestValidStartTime.Year(), closestValidStartTime.Month(), closestValidStartTime.Day(), firstRunHour, firstRunMinute, firstRunSec, 0, time.UTC)
+		if firstRunHour <= closestValidStartTime.Hour() {
+			validStartTime.AddDate(0, 0, 1)
+		}
+		return validStartTime
+
+	case automation.Week:
+		validStartTime := time.Date(closestValidStartTime.Year(), closestValidStartTime.Month(), closestValidStartTime.Day(), firstRunHour, firstRunMinute, firstRunSec, 0, time.UTC)
+		if firstRunDayOfWeek <= int(closestValidStartTime.Weekday()) {
+			dayadd := 7 - (int(closestValidStartTime.Weekday()) - firstRunDayOfWeek)
+			validStartTime.AddDate(0, 0, dayadd)
+		}
+		return validStartTime
+
+	case automation.Month:
+		validStartTime := time.Date(closestValidStartTime.Year(), closestValidStartTime.Month(), firstRunDayOfMonth, firstRunHour, firstRunMinute, firstRunSec, 0, time.UTC)
+		if firstRunDayOfMonth <= closestValidStartTime.Day() {
+			validStartTime.AddDate(0, 1, 0)
+		}
+		return validStartTime
+
+	case automation.OneTime:
+		validStartTime := time.Date(closestValidStartTime.Year(), closestValidStartTime.Month(), firstRunDayOfMonth, firstRunHour, firstRunMinute, firstRunSec, 0, time.UTC)
+		return validStartTime
+
+	}
+
+	log.Printf("[WARN] error compute first valid run time")
+	return closestValidStartTime
+}
+
+func resourceArmAutomationScheduleCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).automationScheduleClient
+	log.Printf("[INFO] preparing arguments for AzureRM Automation Schedule creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	accName := d.Get("account_name").(string)
+	freqstr := d.Get("frequency").(string)
+	freq := automation.ScheduleFrequency(freqstr)
+
+	cst := d.Get("start_time").(string)
+	var starttime time.Time
+
+	if cst != "" {
+		var tperr error
+		starttime, tperr = time.Parse(time.RFC3339, cst)
+		if tperr != nil {
+			return fmt.Errorf("Cannot parse start_time: %s", cst)
+		}
+	} else {
+		starttime = computeValidStartTime(d.Get("first_run").(*schema.Set), freq)
+	}
+
+	ardt := date.Time{Time: starttime}
+
+	//TODO Interval handling currently not supported via Go SDK
+	//Related ticket in the azure go sdk: https://github.com/Azure/azure-sdk-for-go/issues/725
+	//Related ticket in the azure rest api spec: https://github.com/Azure/azure-rest-api-specs/issues/1545
+	//interval :=
+
+	description := d.Get("description").(string)
+	timezone := "UTC"
```

Comment:
can we make this configurable by the user, defaulted to `UTC`?

Reply:
Done, docomentation is also updated